### DISCUSSION TITLE
Is Updating Firmware before flash useful?

### DISCUSSION THREAD

**Discussion by Sn00kiT** [Hypfer/Valetudo#1987] [Q&A / Support] [answered]
Should i update my S5 Max before flashing Valetudo to the newest version over the Xiaomi App and does that make any difference?
Also do i need the fixboot option in Dustbuilder?
I am currently on 3.5.8_1482 and available is 4.1.2_1668 in the Xiaomi App.
I flashed my old S5 a while ago with valetudo and it died on me after receiving the the Leipziger Messer Award :-). So i bought a cheap S5 Max on ebay

**Comment by stefanmd023** [comment, marked as answer]
There is no need to update the firmware, as you're anyway going to place a firmware from dustbuilder, unless otherwise specified.

Don't forget to factory reset the robot!

**Reply by Hypfer** [reply]
Adding to this for people that might stumble upon it during a search:

**Never** update the firmware of your robot via the vendor app if you intend to install Valetudo.
It's always possible that the Vendor will have released an update that blocks our rooting methods.

On the S5 Max it's _probably_ not that important given that it is EOL and thus doesn't seem to receive any updates anymore, but regardless. Don't.

**Reply by Sn00kiT** [reply]
I red the q&a on valetudo, but since it´s EOL i thought the same way. I only thought there might be benefits/bugfixes, because it it is not clearly stated what rooting the device does with the remaing files of the vendor. From my point of view the rooting process only patches the system to add root and functionality (ssh, etc.). But i am unsure if the stripped version completly replaces the system underneath. Never the less i tackled the device today after a nightmare with screws and it works flawlessly :-). Only thing i didn´t get to work is to do it with linux. i always got script errors running the run.sh. i tried first with my main linux mint, but even debian/ubuntu did show the error. i found a little tutorial to do it with windows and this worked indeed without flaws. only the wget part didn´t work. so i just uploaded the valetude arm file via WinSCP to the robot, renamed it and corrected the write permissions. 
Maybe some helps this (it certainly did for me):
https://www.roboter-forum.com/threads/roborock-s5-max-root-valetudo-mit-fel-methode-meine-komplette-anleitung-fuer-anfaenger-und-dummys-wie-mich.56145/
Thank you all for your fast response and the support!

**Comment by Sn00kiT** [comment]
Thank you very much!
